### ISSUE TITLE
MapMouseEvent location incorrect inside UIKit grid

### ISSUE THREAD

**Issue by johankytt** [mapbox/mapbox-gl-js#11550]
**mapbox-gl-js version 2.7.0**:

**Browser: Chrome 98.0.4758.102, Safari 13.1.2**:

### Steps to Trigger Behavior

 1. Create a map inside an UIKit grid, add a point feature
 2. Add the click event listener with feature querying based on the e.point event location
 3. Click on the feature added in step 1
 4. The clicked feature should be returned, but it is not

### Link to Demonstration
https://jsfiddle.net/Johan111/vywdncj0/2/
Check the logs in the console

### Expected Behavior
queryRenderedFeatures(e.point) should return the clicked feature

### Actual Behavior
queryRenderedFeatures(e.point) returns nothing,
but queryRenderedFeatures([e.originalEvent.layerX, e.originalEvent.layerY]) does

It seems to be related to the padding added by the UIKit grid style. The e.point is shifted from the e.originalEvent by the same amount as is the padding.

**Comment by ansis** [comment]
Yep, it looks like the padding. Can you work around the issue by setting padding to zero for the map element? (wrapping it in a different div that does have padding, if you need it)

On our side, we should look into whether we should:
- account for the padding, or
- enforce not having padding
